src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/commands/mobile/subscribe_vehicle_request_test.cc

```diff
@@ -0,0 +1,325 @@
+    ON_CALL(*mock_app_, app_id()).WillByDefault(Return(kConnectionKey));
+  }
+
+  MockAppPtr mock_app_;
+  application_manager::AppExtensionPtr vi_app_extension_ptr_;
+  std::shared_ptr<sync_primitives::Lock> app_set_lock_ptr_;
+  vehicle_info_plugin::VehicleInfoPlugin vi_plugin_;
+  application_manager_test::MockRPCHandler mock_rpc_handler_;
+  resumption::LastStateWrapperPtr mock_last_state_;
+};
+
+TEST_F(SubscribeVehicleRequestTest, Run_AppNotRegistered_UNSUCCESS) {
+  CommandPtr command(CreateCommandVI<SubscribeVehicleDataRequest>());
+  EXPECT_CALL(app_mngr_, application(_))
+      .WillOnce(Return(ApplicationSharedPtr()));
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_result::APPLICATION_NOT_REGISTERED), _));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(SubscribeVehicleRequestTest, Run_DataToSubscribeIsNotExisted_UNSUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg)[am::strings::msg_params][am::strings::button_name] =
+      kVehicleTypeStr;
+
+  am::VehicleData data;
+  EXPECT_CALL(mock_message_helper_, vehicle_data())
+      .WillRepeatedly(ReturnRef(data));
+
+  smart_objects::SmartObject empty_hmi_custom_params;
+  ON_CALL(mock_custom_vehicle_data_manager_, CreateHMIMessageParams(_))
+      .WillByDefault(Return(empty_hmi_custom_params));
+
+  CommandPtr command(CreateCommandVI<SubscribeVehicleDataRequest>(command_msg));
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::INVALID_DATA), _));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(SubscribeVehicleRequestTest,
+       Run_EmptyVehicleData_INVALID_DATA_SentToMobile) {
+  am::VehicleData vehicle_data;
+  EXPECT_CALL(mock_message_helper_, vehicle_data())
+      .WillRepeatedly(ReturnRef(vehicle_data));
+
+  smart_objects::SmartObject empty_hmi_custom_params;
+  ON_CALL(mock_custom_vehicle_data_manager_, CreateHMIMessageParams(_))
+      .WillByDefault(Return(empty_hmi_custom_params));
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::INVALID_DATA), _));
+
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg)[am::strings::msg_params][kMsgParamKey] = false;
+  CommandPtr command(CreateCommandVI<SubscribeVehicleDataRequest>(command_msg));
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(SubscribeVehicleRequestTest, Run_SubscribeDataDisabled_UNSUCCESS) {
```

Comment:
Can't see the place in UT, where data is diabled. Make iot more obvious

Reply:
@LuxoftAKutsan renamed UT so it would better describe its logic. https://github.com/smartdevicelink/sdl_core/pull/3391/commits/2856a9cfbfd40b464cb9bf80a2fdec13dd716f96